docs: accessibility improvements (#4652)
adobe/spectrum-web-components: packages/menu/README.md

</sp-popover>
```

#### Labels

> To render accessibly, an `<sp-menu>` element or its parent `<sp-popover>` must have a label. For an accessible label that is visibly hidden, but read by assistive technology, use the `label` attribute.

Reviewer: ```suggestion
To render accessibly, an `<sp-menu>` element or its parent `<sp-popover>` must have a label. For an accessible label that is visibly hidden, but can still be read by assistive technology, use the `label` attribute.
```

    <sp-menu-item>Deselect</sp-menu-item>
    <sp-menu-item>Select inverse</sp-menu-item>
    <sp-menu-item>Feather...</sp-menu-item>
    <sp-menu-item>Select and mask...</sp-menu-item>
    <sp-menu-divider></sp-menu-divider>
    <sp-menu-item>Save selection</sp-menu-item>
    <sp-menu-item disabled>Make work path</sp-menu-item>
</sp-menu>
<sp-popover open style="position: relative" label="Selection type:">
    <sp-menu id="menu-xl-popover" size="xl">
        <sp-menu-item>Deselect</sp-menu-item>
        <sp-menu-item>Select inverse</sp-menu-item>
        <sp-menu-item>Feather...</sp-menu-item>
        <sp-menu-item>Select and mask...</sp-menu-item>
        <sp-menu-divider></sp-menu-divider>
        <sp-menu-item>Save selection</sp-menu-item>
        <sp-menu-item disabled>Make work path</sp-menu-item>
    </sp-menu>
</sp-popover>
```

</sp-tab-panel>
</sp-tabs>

#### Selection

The `<sp-menu>` element can be instructed to maintain a selection via the `selects` attribute. Depending on the chosen algorithm, the `<sp-menu>` element will hold a `value` property and manage the `selected` state of its `<sp-menu-item>` descendants.

> -   When `selects` is set to `single`, the `<sp-menu>` element will maintain one selected item after an initial selection is made.

Reviewer: ```suggestion
-   When `selects="single"`, the `<sp-menu>` element will maintain one selected item after an initial selection is made.
```

What if we did this? I don't have a strong opinion on either way.

</sp-menu>
```

</sp-tab-panel>
</sp-tabs>

### Behaviors

#### "change" event

> Whether `<sp-menu>` carries a selection or not, when one of the `<sp-menu-item>` children that it manages is activated the `<sp-menu>` element will dispatch a `change` event. At dispatch time, even when a selection is not held due to the absence of the `selects` attribute, the `value` of the `<sp-menu>` will correspond to the `<sp-menu-item>` that was activated. When the `selects` attribute is present, this `value` will be persisted beyond the lifecycle of the `change` event. When `selects="multiple"` the values of multiple items will be comma separated unless otherwise set via the `value-separator` attribute.

Reviewer: ```suggestion
Regardless of whether or not `<sp-menu>` carries a selection, when one of the `<sp-menu-item>` children that it manages is activated, the `<sp-menu>` element will dispatch a `change` event. At dispatch time, even when a selection is not held due to the absence of the `selects` attribute, the `value` of the `<sp-menu>` will correspond to the `<sp-menu-item>` that was activated. When the `selects` attribute is present, this `value` will persist beyond the lifecycle of the `change` event. When `selects="multiple"`, the values of multiple items will be comma separated, unless otherwise set via the `value-separator` attribute.
```
Are the "values" referred to in this statement referring to the actual `value` attribute?

Note: The `change` event is only dispatched on a left mouse click or Enter/Space keypress. Right/Middle mouse clicks will not dispatch the `change` event.

### Accessibility guidelines

#### Include a label

> In rare cases where context is sufficient and a label could be absent, make sure to have the design reviewed and approved by an accessibility expert.

Reviewer 2: Not sure I can think of a reason not to have a label for a menu as it is not rendered.

> In rare cases where context is sufficient and a label could be absent, make sure to have the design reviewed and approved by an accessibility expert.

Reviewer: ```suggestion
In rare cases where context is sufficient enough that a label could be absent, make sure to have the design reviewed and approved by an accessibility expert.
```
